core-js/src/main/javascript/cdf/components/ccc/BaseCccComponent.js

```diff
@@ -68,49 +163,81 @@ define([
       var me = this;
 
       // Set up defaults for height and width
-      if(typeof(this.chartDefinition.width) === "undefined") {
+      if (typeof(this.chartDefinition.width) === "undefined") {
         this.chartDefinition.width = ph.width();
       }
 
-      if(typeof(this.chartDefinition.height) === "undefined") {
+      if (typeof(this.chartDefinition.height) === "undefined") {
         this.chartDefinition.height = ph.height();
       }
 
-      if(typeof Modernizr !== 'undefined' && Modernizr.svg) {
+      if (typeof Modernizr !== 'undefined' && Modernizr.svg) {
         this.renderChart();
       } else {
-        pv.listenForPageLoad(function() {
+        pv.listenForPageLoad(function () {
           me.renderChart();
         });
       }
     },
 
-    render: function(values) {
+    render: function (data) {
+      this._renderOuter(data);
+    },
+    
+    _renderOuter: function (data) {
+      var promiseExtensions;
+
+      this._preProcessChartDefinition();
 
+      if (this._shouldUseVizApiStyles()) {
+        promiseExtensions = _context.getAsync('pentaho/ccc/visual/' + this.getCccChartName())
+            .then(function (View) {
+              return View.type.extensionEffective;
+            });
+      } else {
+        // no external extensions
+        promiseExtensions = Promise.resolve(null);
+      }
+
+      promiseExtensions
+          .then(_.bind(this._renderInner, this, data))
+          .then(_.bind(this.endExec, this), _.bind(this.failExec, this));
+    },
+    
+    _renderInner: function (data, externalChartDefinition) {
       $("#" + this.htmlObject).append('<div id="' + this.htmlObject + 'protovis"></div>');
 
-      this._preProcessChartDefinition();
+      // Always clone the original chartDefinition.
+      var cd = $.extend({}, this.chartDefinition);
+
+      if (externalChartDefinition) {
+        cd = $.extend(externalChartDefinition, cd);
```

Comment:
I know I wrote this bit... but it is intended that chartDefinition prevails right? `externalChartDefinition` are default values. Just checking.